lib/fathead/python_pymongo/parse.py

```diff
@@ -0,0 +1,282 @@
+        """
+        self.baseurl = baseurl
+        self.downloaddir = downloaddir
+        self.api_files = []
+        self.get_api_doc_files()
+        self.output = ""
+        self.print_output()
+
+    def get_api_doc_files(self):
+        """
+        Reads index.html of downloaded API docs and parses for links to api docs files
+        """
+        self.api_files = []
+        entry_html = open(self.downloaddir + 'index.html')
+        soupy = BeautifulSoup(entry_html, 'html.parser')
+        for element in soupy.find_all("a", {"class": "reference internal", "href": re.compile('^.*\.html$')}):
+            self.api_files.append(self.downloaddir + element["href"])
+        del soupy
+
+    def print_output(self):
+        """
+        Parses found files, prints formatted output for DDG Fathead to stdout.
+        """
+        for filename in self.api_files:
+            html_data = open(filename).read()
+            soupy = BeautifulSoup(html_data, 'html.parser')
+            # Get any classes in the file
+            for element in soupy.find_all("dl", {"class": "class"}):
+                anchor_link = ""
+                if element.find("a", {"class": "headerlink"}):
+                    anchor_link = element.find("a", {"class": "headerlink"})["href"]
+                code = ""
+                description = ""
+                # Format methods/classmethods/functions/attributes/exceptions within a class
+                tags_to_replace = ['method', 'classmethod', 'function', 'attribute', 'exception', 'describe']
+                for tag_to_replace in tags_to_replace:
+                    for tag in element.find_all("dl", {"class": tag_to_replace}):
+                        tag_code = ""
+                        tag_description = ""
+                        for item in tag.dt.contents:
+                            tag_code += str(item)
+                        tag_code = self._clean_html_tags(tag_code)
+                        for item in tag.dd.contents:
+                            tag_description += str(item)
+                        tag_description = self._clean_html_tags(tag_description)
+                        description += tag_code
+                        description += tag_description
+                        tag.decompose()
+                output_line = ""
+                for item in element.dt.contents:
+                    code += str(item)
+                code = self._clean_html_tags(code)
+                code = self._clean_code_tags(code)
+                for item in element.dd.contents:
+                    description += str(item)
+
+                description = self._clean_html_tags(description)
+
+                abstract = '<section class="prog_container">' + \
+                           '<pre><code>' + code + '</pre></code><p>' + \
+                           description + '</p></section>'
+                title = element.dt["id"].replace('.', ' ')
```

Comment:
@dankolbrs it would probably be best to maintain the original `id` as the title, but additionally creating a redirect where the `.` is replaced with a space would be good.

That way "bson.BSON" will still trigger, and show the result you expect, as well as "bson BSON". The title of the result will also be displayed correctly though, showing "bson.BSON"